frontend/src/app/questions/questions.component.ts

```diff
@@ -0,0 +1,314 @@
+    ) {}
+
+    ngOnInit() {
+        // two way binding for forms is not working for some reason unless question is initialised with empty values
+        this.initQuestion();
+
+        // fetch data from API call
+        this.handleInitData();
+
+        this.initDifficulties();
+
+        this.initFormGroup();
+
+        this.initListeners();
+    }
+
+    openNewQuestion() {
+        this.resetFormGroup();
+        this.question = {} as Question;
+        this.submitted = false;
+        this.isDialogVisible = true;
+    }
+
+    deleteSelectedQuestions() {
+        this.confirmationService.confirm({
+            message: 'Are you sure you want to delete the selected questions?',
+            header: 'Delete Confirmation',
+            icon: 'pi pi-exclamation-triangle',
+            accept: () => {
+                this.handleDeleteQuestionResponse();
+            },
+        });
+    }
+
+    saveQuestion() {
+        this.submitted = true;
+
+        if (
+            !this.question.title?.trim() ||
+            !this.question.topics ||
+            this.question.topics?.length == 0 ||
+            !this.question.difficulty?.trim() ||
+            !this.question.description?.trim()
+        ) {
+            return;
+        }
+
+        if (this.question.id) {
+            // update
+            const { id, ...questionBody } = this.question;
+            this.handleEditQuestionResponse(id, questionBody);
+        } else {
+            // add
+            this.handleAddQuestionResponse();
+        }
+
+        this.isDialogVisible = false;
+        this.question = {} as Question;
+    }
+
+    resetFormGroup() {
+        this.questionFormGroup.reset({
+            selectedTopics: [],
+            selectedDifficulty: '',
+            textTitle: '',
+            textDescription: '',
+        });
+    }
+
+    editQuestion(question: Question) {
+        this.question.id = question.id;
+        this.questionFormGroup.patchValue({
+            textTitle: question.title,
+            textDescription: question.description,
+            selectedTopics: question.topics,
+            selectedDifficulty: question.difficulty,
+        });
+        this.isDialogVisible = true;
+    }
+
+    initListeners() {
+        // Dropdown difficulty listener
+        this.questionFormGroup.get('selectedDifficulty')?.valueChanges.subscribe(v => {
+            this.question.difficulty = v;
+        });
+
+        // Multiselect topics listener
+        this.questionFormGroup.get('selectedTopics')?.valueChanges.subscribe(v => {
+            this.question.topics = v;
+        });
+
+        // text title listener
+        this.questionFormGroup.get('textTitle')?.valueChanges.subscribe(v => {
+            this.question.title = v;
+        });
+
+        // text description listener
+        this.questionFormGroup.get('textDescription')?.valueChanges.subscribe(v => {
+            this.question.description = v;
+        });
+    }
```

Comment:
Not sure why this is required when we can get all fields of the form at any time with `this.questionFormGroup.value`.

Reply:
Yup agreed. This could be replaced with `this.questionFormGroup.value`.